code/modules/projectiles/boxes_magazines/_box_magazine.dm

```diff
@@ -35,18 +35,42 @@
 
 /obj/item/ammo_box/Initialize()
 	. = ..()
+	// We do it this way because box
 	if (!bullet_cost)
-		for (var/material in custom_materials)
-			var/material_amount = custom_materials[material]
-			LAZYSET(base_cost, material, (material_amount * 0.10))
-
-			material_amount *= 0.90 // 10% for the container
-			material_amount /= max_ammo
-			LAZYSET(bullet_cost, material, material_amount)
+		_refresh_bullet_cost()
 	if(!start_empty)
 		top_off(starting=TRUE)
 	update_icon()
 
+/**
+  * Called to recalculate the matinal cost of a bullet
+  *
+  * Should only be called when the custom_materials is changed
+  * so usually on create
+  *
+  */
+/obj/item/ammo_box/proc/_refresh_bullet_cost()
+	var/ammo_count = stored_ammo?.len || max_ammo
+	bullet_cost = list()
+	base_cost = list()
+	for (var/material in custom_materials)
+		var/material_amount = custom_materials[material]
+		base_cost[material] = (material_amount * 0.10)
+		material_amount *= 0.90 // 10% for the container
+		material_amount /= ammo_count
+		bullet_cost[material] = material_amount
+	if(length(stored_ammo))
+		for(var/atom/A in stored_ammo) // hard set all the bullets to this cost
```

Comment:
Can `stored_ammo` hold anything that's not an `/atom`? If not, then there's no need to type check here.

Reply:
It shouldn't.  Is it slower than casting it though?  I thought all this does is just loop through a list and skip non typed?  I still need to cast it as an atom to set its custom_material cost

Reply:
Prefer to use something like this:
```
for(var/a in stored_ammo)
  var/atom/ammo = a
```
This has two advantages:
1) It is faster, as it doesn't have to typecheck every loop. Since we know the contents of the list there's no need for checking.
2) In case our list gets polluted by `nulls`, more often than not for improperly handled reference cleaning on deletion, it will runtime and warn us about it. Nulls in list are pain, and we should know about them.